[Slack MCP] adding missing scope and improving output + flow

LeandreLeBizec opened the pull request:
## Description

This PR improves the Slack MCP semantic search implementation:

**Missing scopes added**: The `channel_types` parameter is now explicitly passed to the semantic search API, and missing OAuth scopes (`search:read.im`, `search:read.mpim`) have been added to enable proper searching across all conversation types.

**ID and user name and channel resolution**: Previously, deleted users or users external to the workspace would return "unknown" as the mention display in user-facing output, which provided poor usability. Additionally, with the introduction of DMs and group DMs—which have different name formatting than channels—these were also displayed as "unknown".
This PR addresses these issues by always retrieving and displaying the ID alongside the name, or in place of the name when unavailable.
Benefits:
- Immediate usability: Users can identify the referenced user/channel even when the name is unavailable
- Reduced API calls: When a user finds the desired message, they don't need to make an additional call to retrieve the user ID or channel ID for subsequent processing with the agent
- Consistency: All mention types (workspace users, external users, channels, DMs, group DMs) now have predictable, actionable output
- This enables seamless chaining of agent operations without requiring intermediary lookup steps.

## Risk

OAuth scope changes require users to reconnect their Slack account. The new scopes (`search:read.im`, `search:read.mpim`) grant access to DMs and group DMs, which users might perceive as more invasive.

## Tests

Manual testing with whitelisted Slack app to verify semantic search works across all channel types and that search results include proper user/channel ID resolution.

## Deploy Plan
Run Front

LeandreLeBizec committed:
[Slack MCP] adding missing scope and improving output + flow

LeandreLeBizec committed:
update promise.all by concurrentExecutor

spolu commented on front/lib/actions/mcp_internal_actions/servers/slack/slack_personal.ts:
```diff
@@ -114,37 +157,55 @@ export const slackSearch = async (
+  userIdToName: Map<string, string | null>;
+  channelIdToName: Map<string, string | null>;
+}> {
+  const uniqueUserIds = [
```
how many in general?

spolu commented on front/lib/actions/mcp_internal_actions/servers/slack/slack_personal.ts:
```diff
@@ -114,37 +157,55 @@ export const slackSearch = async (
+      matches.map((m) => m.user_id).filter((id): id is string => !!id)
+    ),
+  ];
+  const uniqueChannelIds = [
```
how many in general?

spolu commented on front/lib/actions/mcp_internal_actions/servers/slack/slack_personal.ts:
```diff
@@ -547,36 +636,51 @@ async function createServer(
+                        `@${username}`
+                    );
+
+                    return JSON.stringify(
```
Hmm this is very token intensive. Can we use a more compact presentation?

LeandreLeBizec committed:
Merge remote-tracking branch 'origin/main' into leandre/slack-mcp-fix-search-message

LeandreLeBizec committed:
reverse json for plain text in search results

LeandreLeBizec committed:
lint

LeandreLeBizec commented on front/lib/api/oauth/providers/slack.ts:
```diff
@@ -93,6 +95,7 @@ export class SlackOAuthProvider implements BaseOAuthStrategyProvider {
             "im:read",
             "users:read",
             "users:read.email",
+            "search:read.public",
```
**Why `getAIEnablementStatus` uses `platform_actions` scopes (not `personal_actions`)** 

The MCP Slack connection flow works as follows:
1. Initial setup (admin): When admins configure the MCP server ([CreateMCPServerSheet:L219](https://github.com/dust-tt/dust/blob/main/front/components/actions/mcp/CreateMCPServerSheet.tsx#L219)), the code always forces `platform_actions` as the useCase, generating a workspace-level bot token (xoxb-...) that's stored in our OAuth DB.
2. Scope definition: The OAuth URL (built in [slack.ts:L147-165](https://github.com/dust-tt/dust/blob/main/front/lib/api/oauth/providers/slack.ts#L147-165) passes scopes defined in the `platform_actions` case. These scopes are included in the authorization URL's scope= parameter.
3. Unexpected behavior: Through testing, I found that scopes passed in the OAuth URL are granted regardless of what's in the app manifest. For example, adding `search:read.public` to `platform_actions` works even though it's not in our manifest. This might be because our whitelisted dev app (`Dust-dev-whitelisted`) has looser restrictions in development mode, but I couldn't find official documentation confirming this.
4. Empirical proof: 
With `search:read.public` in `platform_actions`: `assistant.search.info` works. 
Without it: Returns `missing_scope` error with needed: `search:read.public`
5. Usage: `getAIEnablementStatus` in [slack_personal.ts:L338](https://github.com/dust-tt/dust/blob/main/front/lib/actions/mcp_internal_actions/servers/slack/slack_personal.ts#L338) uses the bot token generated during initial setup with `platform_actions` scopes.

Why can we add scopes "on the fly" via the OAuth URL even if they're not in the manifest? If you have any clue on that I'm listening. Is this specific to whitelisted dev apps? I couldn't find documentation on this behavior. I can ask directly on the share slack channel we have with slack.

davidebbo replied:
That makes no sense that we could be granted permissions that are not in our manifest, because the user would not have agreed to them. The dialog they get specifically lists all the things that we ask for.

Note that technically, I think what matters is "what was in our manifest at the time the did the oauth flow", more than "what's in our manifest right now". Because even if we change the manifest, we keep the old token based on the previous manifest. But that doesn't solve this mystery.

LeandreLeBizec replied:
@davidebbo Completely aligned. Here is the [link](https://dust4ai.slack.com/archives/C050SM8NSPK/p1764240781952659?thread_ts=1764240586.853989&cid=C050SM8NSPK) of the video I made to demonstrate what is happening. Adding the link here for clear context.

LeandreLeBizec replied:
@spolu 
This impacts the `platform_actions` case which uses the official `Dust` Slack app whitelisted on the Slack store that is used by Dust's Slack integration (is that what you call `bot`?) and Frank's `slack_bot` MCP under feature flag.

What might need to be changed at this point is:
- the case that gets called when adding an MCP server ('tool' in the UI) in Dust. This is what's done [here](https://github.com/dust-tt/dust/blob/main/front/components/actions/mcp/CreateMCPServerSheet.tsx#L219). It's always `platform_action` during setup, which means it's always the official `Dust` Slack app that will be called during setup.
- The way we are choosing between keyword search and semantic search for the user. But that would need additional discussions.

davidebbo replied:
Hopefully, my [new table](https://www.notion.so/dust-tt/Runbook-Slack-25e28599d9418003bea0f51f181621b9?source=copy_link) helps discuss this without confusing ourselves. So yes, this is the bot app.

> That makes no sense that we could be granted permissions that are not in our manifest, because the user would not have agreed to them. The dialog they get specifically lists all the things that we ask for.

So now I see I was confused here. The dialog reflects what the code asks for, and not what's in the manifest. But there is expectation that the code can only ask a subset of what's in the manifest, and @LeandreLeBizec has established that it's not the case (making the manifest scopes seemingly irrelevant). We do not understand this part.

Side note: there are a lot of common things between the 3 lists. It may be good to have a core list shared by all, and then add the delta for each, to highlight the differences.

LeandreLeBizec committed:
Merge main into leandre/slack-mcp-fix-search-message

Resolved conflict in front/lib/api/oauth/providers/slack.ts by accepting main's version (personal_actions case removed, keeping only assertion).

LeandreLeBizec committed:
missing scopes for DMs and lint

LeandreLeBizec committed:
Merge branch 'main' of github.com:dust-tt/dust into leandre/slack-mcp-fix-search-message

LeandreLeBizec committed:
minimum code to semantic search in private channels, dms and group dms

LeandreLeBizec committed:
helper function user in semantic search and keyword search

LeandreLeBizec committed:
change name for id in slack response

some users and channels are not correctly displayed (deleted users, external users, DMs or group DMs and Unknown is printed in the output. It's not very usable. ID is solving this problem. Next step is ID resolution to print name, and ID if no name

LeandreLeBizec committed:
name and channel resolution

LeandreLeBizec committed:
lint

LeandreLeBizec committed:
missing :

LeandreLeBizec committed:
coding rule violations

LeandreLeBizec committed:
Revert package-lock.json changes

davidebbo commented on front/lib/actions/mcp_internal_actions/servers/slack/slack_personal.ts:
```diff
@@ -85,19 +88,38 @@ export const slackSearch = async (
+        channel_name: msg.channel_name,
+        message_ts: msg.message_ts,
+        content: msg.content,
+        permalink: msg.permalink,
```
Same here ideally.

davidebbo commented on front/lib/actions/mcp_internal_actions/servers/slack/slack_personal.ts:
```diff
@@ -146,6 +170,33 @@ export const slackSearch = async (
+// Cleans up Slack-specific formatting and returns a human-readable string.
+function formatSlackMessageForDisplay(match: SlackSearchMatch): string {
+  const author = match.author_name
+    ? `${match.author_name} (${match.author_id ?? ""})`
```
This would end up with `()` if author_id is null. Is that ok? Same for channel.

davidebbo reviewed (approved):
LGTM, with a few comments

LeandreLeBizec committed:
type SlackSearchResponse and parenthesis issue if author_id is null
